pyirf/irf/energy_dispersion.py

```diff
@@ -80,6 +84,56 @@ def energy_dispersion(
     return energy_dispersion
 
 
+@u.quantity_input(true_energy_bins=u.TeV, reco_energy_bins=u.TeV, fov_offset_bins=u.deg)
+def energy_migration_matrix(
+    events, true_energy_bins, reco_energy_bins, fov_offset_bins
+):
+    """Compute the energy migration matrix directly from the events.
+
+    Parameters
+    ----------
+    events : `~astropy.table.QTable`
+        Table of the DL2 events.
+        Required columns: ``reco_energy``, ``true_energy``, ``true_source_fov_offset``.
+    true_energy_bins : `~astropy.units.Quantity`
+        Bin edges in true energy.
+    reco_energy_bins : `~astropy.units.Quantity`
+        Bin edges in reconstructed energy.
+
+    Returns
+    -------
+    matrix : array-like
+        Migration matrix as probabilities along the true
```

Comment:
The probabilities are along the reco energy axis. The migration matrix maps from true energy to the reco energies.

Reply:
fixed